src/sdk/namespace_fs.js

```diff
@@ -424,26 +539,37 @@ class NamespaceFS {
                  */
                 const process_entry = async ent => {
                     // dbg.log0('process_entry', dir_key, ent.name);
-                    if (!ent.name.startsWith(prefix_ent) ||
+                    if ((!ent.name.startsWith(prefix_ent) ||
                         ent.name < marker_curr ||
                         ent.name === this.get_bucket_tmpdir() ||
-                        ent.name === config.NSFS_FOLDER_OBJECT_NAME) {
+                        ent.name === config.NSFS_FOLDER_OBJECT_NAME) &&
+                        !this._is_hidden_version_path(ent.name)) {
                         return;
                     }
-
                     const isDir = await is_directory_or_symlink_to_directory(ent, fs_context, path.join(dir_path, ent.name));
 
-                    const r = {
-                        key: this._get_entry_key(dir_key, ent, isDir),
-                        common_prefix: isDir,
-                    };
+                    let r;
+                    if (list_versions && ent.name.includes('_mtime-')) {
```

Comment:
it's a bit overkill but in case the object name includes '_mtime-' and it's not a version we fail here